### PR TITLE
FIX: Prevent log probabilities from turning to infinite

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR adds the same fix as in https://github.com/uxlfoundation/oneDAL/pull/3293 for the prediction function that outputs log probabilities, so that sklearnex's `predict_log_proba` doesn't end up returning infinites.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
I was able to reproduce the issue on the same machine where the CI error was reported. It appears to be an issue of not checking for null pointers or error codes somwhere, most likely on the daal4py side:
```
==3694450== Invalid read of size 8
==3694450==    at 0x42035D4: UnknownInlinedFun (pycore_pystate.h:133)
==3694450==    by 0x42035D4: UnknownInlinedFun (obmalloc.c:866)
==3694450==    by 0x42035D4: UnknownInlinedFun (obmalloc.c:1850)
==3694450==    by 0x42035D4: PyObject_Free (obmalloc.c:830)
==3694450==    by 0x3EF89483: Py_DECREF (object.h:705)
==3694450==    by 0x3EF89483: Py_XDECREF (object.h:798)
==3694450==    by 0x3EF89483: ~NpyNumericTable (npy4daal.h:372)
==3694450==    by 0x3EF89483: NpyNumericTable<NpyNonContigHandler>::~NpyNumericTable() (npy4daal.h:372)
==3694450==    by 0x3F5EBEBF: _remove (daal_shared_ptr.h:352)
==3694450==    by 0x3F5EBEBF: operator= (daal_shared_ptr.h:366)
==3694450==    by 0x3F5EBEBF: daal::algorithms::interface1::Argument::set(unsigned long, daal::services::interface1::SharedPtr<daal::data_management::interface1::SerializationIface> const&) (algorithm_base_impl.cpp:73)
==3694450==    by 0x3F5F7BB5: daal::algorithms::classifier::prediction::interface1::Input::set(daal::algorithms::classifier::prediction::NumericTableInputId, daal::services::interface1::SharedPtr<daal::data_management::interface1::NumericTable> const&) (classifier_predict.cpp:92)
==3694450==    by 0x3ECDF60E: logistic_regression_prediction_manager<double, (daal::algorithms::logistic_regression::prediction::Method)0>::batch(bool) (daal4py_cpp.h:5609)
==3694450==    by 0x3ECDF51A: logistic_regression_prediction_manager<double, (daal::algorithms::logistic_regression::prediction::Method)0>::compute(data_or_file const&, daal::services::interface1::SharedPtr<daal::algorithms::logistic_regression::interface1::Model>*, bool) (daal4py_cpp.h:5630)
==3694450==    by 0x3EE7AA4C: __pyx_pf_7daal4py_8_daal4py_30logistic_regression_prediction_4_compute (daal4py_cy.cpp:205893)
==3694450==    by 0x3EE7AA4C: __pyx_pw_7daal4py_8_daal4py_30logistic_regression_prediction_5_compute(_object*, _object* const*, long, _object*) (daal4py_cy.cpp:205711)
==3694450==    by 0x4217165: UnknownInlinedFun (pycore_call.h:92)
==3694450==    by 0x4217165: PyObject_VectorcallMethod (call.c:887)
==3694450==    by 0x3EE7B178: __pyx_pf_7daal4py_8_daal4py_30logistic_regression_prediction_6compute (daal4py_cy.cpp:206102)
==3694450==    by 0x3EE7B178: __pyx_pw_7daal4py_8_daal4py_30logistic_regression_prediction_7compute(_object*, _object* const*, long, _object*) (daal4py_cy.cpp:206051)
==3694450==    by 0x4233CEE: UnknownInlinedFun (pycore_call.h:92)
==3694450==    by 0x4233CEE: PyObject_Vectorcall (call.c:325)
==3694450==    by 0x421CDA8: _PyEval_EvalFrameDefault (bytecodes.c:2715)
==3694450==    by 0x4214BCA: _PyObject_FastCallDictTstate (call.c:144)
==3694450==  Address 0x10 is not stack'd, malloc'd or (recently) free'd
==3694450== 
Fatal Python error: Segmentation fault
```

Doesn't appear to be related to the changes here though. Will investigate further.

**Comment by david-cortes-intel** [comment]
Will need to merge this PR before in order for the CI jobs to pass:
https://github.com/uxlfoundation/scikit-learn-intelex/pull/2665
